### PR TITLE
login customer immediately

### PR DESCRIPTION
**Pull request by piyush-getsimpl**
Force login (only for simpl checkout) the newly created customer when guest order is disabled.

### CODE REVIEW

File: includes/helpers/simpl_login_helper.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		// Force login only when guest checkout is not allowd and we create the customer
+		if('yes' !== get_option( 'woocommerce_enable_guest_checkout' )) {
+
+			$order_id = wc_get_order_id_by_order_key( $_GET['key'] );
+			$order = wc_get_order( $order_id );
+
+			// Force login only for Simpl checkout
+			if('yes' == $order->get_meta(SIMPL_ORDER_METADATA) ) {			
+				
+				$user_id = $order->get_customer_id();
```
Comment (Chhekur) [inline includes/helpers/simpl_login_helper.php:20]: I think this should be on top, if this is not a simpl order then we shouldn't event touch it just return, thoughts?

**Review by Chhekur — COMMENTED** [review]
let's add TODO for adding a check if user is already logged in while our customer_creation flow.

File: includes/helpers/simpl_login_helper.php (hunk 1 above)
Comment (piyush-getsimpl) [reply includes/helpers/simpl_login_helper.php:20]: Debatable - The thought behind keeping it this way is as below:
- This flow runs for all redirects and hence keeping 'order-received' check at the top
- All the native and even more than 50% of our orders would be guest checkout and hence second priority
- Lastly checking for simpl orders
If 'woocommerce_enable_guest_checkout' is being pulled from cache, this should not be latent. If not, this should be put last.

**Comment by piyush-getsimpl** [comment]
> let's add TODO for adding a check if user is already logged in while our customer_creation flow.

Yes, missed it. Let me add.

File: includes/helpers/simpl_login_helper.php (hunk 1 above)
Comment (piyush-getsimpl) [reply includes/helpers/simpl_login_helper.php:20]: Also, to check simpl order, we need to fetch order - so keeping it last to avoid DB call.

**Review by Chhekur — APPROVED** [review]

**Comment by piyush-getsimpl** [comment]
> > let's add TODO for adding a check if user is already logged in while our customer_creation flow.
> 
> Yes, missed it. Let me add.

Done!

**Review by Chhekur — APPROVED** [review]

File: includes/helpers/simpl_login_helper.php (hunk 1 above)
Comment (piyush-getsimpl) [reply includes/helpers/simpl_login_helper.php:23]: But we need order to find if it is Simpl order

**Pull request merged**